internal/btf/btf.go

```diff
@@ -593,43 +581,8 @@ func (h *Handle) FD() int {
 
 // Map is the BTF for a map.
 type Map struct {
```

Comment:
* We could move this type to `ebpf` and hide the fields instead. What would we call the type? `ebpf.MapBTF`?
* We could embed these fields in `MapSpec` directly.

Reply:
The first option sounds right to me. The type doesn't seem to have much to do with other types in the `btf` package. `ebpf.MapBTF` :+1:

Reply:
Why would `btf.Map` be moved and `btf.Program` be kept in place?

Reply:
I'd say we keep it as it is now then?